h2o-algos/src/main/java/hex/glm/RegressionInfluenceDiagnosticsTasks.java

```diff
@@ -0,0 +1,321 @@
+ * http link.
+ */
+public class RegressionInfluenceDiagnosticsTasks {
+  public static class RegressionInfluenceDiagBinomial extends MRTask<RegressionInfluenceDiagBinomial> {
+    final double[] _beta;
+    final double[][] _gramInv;  // could be with standardized or non-standardized predictors, not scaled with obj_reg
+    final Job _j;
+    final int _betaSize;
+    final int _reducedBetaSize;
+    final GLMModel.GLMParameters _parms;
+    final DataInfo _dinfo;
+    final double[] _stdErr;
+    final boolean _foundRedCols;
+    final double[] _oneOverStdErr;
+
+    public RegressionInfluenceDiagBinomial(Job j, double[] beta, double[][] gramInv, GLMModel.GLMParameters parms, 
+                                           DataInfo dinfo, double[] stdErr) {
+      _j = j;
+      _beta = beta; // denormalized beta
+      _betaSize = beta.length;
+      _reducedBetaSize = gramInv.length;
+      _foundRedCols = !(_betaSize == _reducedBetaSize);
+      _gramInv = gramInv; // not scaled by parms._obj_reg
+      _parms = parms;
+      _dinfo = dinfo;
+      _stdErr = stdErr;
+      _oneOverStdErr = Arrays.stream(_stdErr).map(x -> 1.0/x).toArray();
+      
+    }
+
+    @Override
+    public void map(Chunk[] chks, NewChunk[] nc) {
+      if (isCancelled() || _j != null && _j.stop_requested()) return;
+      double[] dfbetas = new double[_betaSize];
+      double[] dfbetasReduced = new double[_reducedBetaSize];
+      double[] row2Array = new double[_betaSize];
+      double[] row2ArrayReduced = new double[_reducedBetaSize];
+      double[] xTimesGramInv = new double[_reducedBetaSize];
+      DataInfo.Row r = _dinfo.newDenseRow();
+      for (int rid = 0; rid < chks[0]._len; ++rid) {
+        _dinfo.extractDenseRow(chks, rid, r);
+        genDfBetasRow(r, nc, row2Array, row2ArrayReduced, dfbetas, dfbetasReduced, xTimesGramInv);
+      }
+
+      if (_j != null)
+       _j.update(1);
+    }
+
+    private void genDfBetasRow(DataInfo.Row r, NewChunk[] nc, double[] row2Array, double[] row2ArrayRed, 
+                               double[] dfbetas, double[] dfbetasRed, double[] xTimesGramInv) {
+      if (r.response_bad) {
+        Arrays.fill(dfbetas, Double.NaN);
+      } else if (r.weight == 0) {
+        Arrays.fill(dfbetas, 0.0);
+      } else {
+        r.expandCatsPredsOnly(row2Array);  // change Row to array
+        if (_foundRedCols) {
+          removeRedCols(row2Array, row2ArrayRed, _stdErr);
+          genDfBeta(r, row2ArrayRed, xTimesGramInv, dfbetasRed, nc);
+        } else {
+          genDfBeta(r, row2Array, xTimesGramInv, dfbetas, nc);
+        }
+      }
+    }
+    
+    private void genDfBeta(DataInfo.Row r, double[] row2Array, double[] xTimesGramInv, double[] dfbetas, NewChunk[] nc) {
+      double mu = _parms.linkInv(r.innerProduct(_beta)+r.offset); // generate p hat
+      // generate residual
+      double residual = r.response(0)-mu; // y-pi, I don't think this is right.  Should be 1-mu when y=1, mu-1 when y=0
```

Comment:
It looks reasonable to me to have `y-mu`.
My *intuition* might be wrong but I'll try to explain it here:

We have loss function that looks something like `-sum(y*ln(mu) + (1-y)*ln(1-mu))` so the closer the prediction is to the correct value the smaller loss so in turn smaller influence on betas (extreme case: we predict `mu` perfectly so we don't need to change betas at all so the influence of that one row on betas is 0).

Now when we look at the `genDfBetas`, the residual multiplies something that does not depend on `y`. So in order to have the effect of closer prediction having smaller impact on betas we need to incorporate the information there somehow.

If we would use `1-mu` and `mu-1` for `y==1` and `y==0` respectively, it would do something counterintuitive to me, e.g., let's assume `mu == 0.25`, if `y==1` => `residual = 0.75`, and if `y==0` => `residual =-0.75`. So the magnitude of the influence would be the same but the loss for that particular row would be different: `ln(0.25)` vs `ln(0.75)`.

Reply:
Your instinct is correct.  This term comes from taking the derivative of the loss function with respect to mu.